Fix to compile with Visual C++ and /Zc:implicitNoexcept-.

jaykrell opened the pull request:
Otherwise gets:
  Error C2694 'override': overriding virtual function
  has less restrictive exception specification than base class virtual
  member function 'base'

Similar changes are being made e.g.:
  https://github.com/boostorg/json/pull/636

See also, similar from me:
https://github.com/boostorg/format/pull/85

jaykrell commented:
Ok. Serious question then: instead of `= default` should I use `{ }`?
Or is that going to "too old" compiler?

rdoeffinger commented:
I think you should squash the 3 commits into a single one, as it's not desirable to have not-quite-correct versions in the history.
Note: I am not speaking with any particular authority, just making a drive-by comment...

jaykrell commented:
> I think you should squash the 3 commits into a single one, as it's not desirable to have not-quite-correct versions in the history. Note: I am not speaking with any particular authority, just making a drive-by comment...

I agree, but, is it approved othewise? That can be done upon commit in the GitHub gui, including editing the commit message.

Opinions vary and are strong about PR workflow.
I've seen people, reviewing my code, request strongly never to rebase a PR, as it makes the PR harder to read.
Even if then squash upon commit is the only choice.

mclow commented:
This looks OK to me.

jaykrell commented:
@mclow I squashed and rebased, ok?
(Is squash really required? A lot of repositories are configured to do that upon commit, not optionally.
Imho that is 99% the right policy. Unless maybe CI runs and passes at each commit, and users and reviewers agree the history reads ok/better with multiple commits. But I've never seen this.)

pdimov commented:
Closing/reopening to retrigger CI.
